core/trino-main/src/main/java/io/trino/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -113,8 +113,10 @@
  */
 class AggregationAnalyzer
 {
+    private final List<Expression> groupByExpressions;
     // fields and expressions in the group by clause
-    private final Set<FieldId> groupingFields;
+    @Nullable
+    private Set<FieldId> groupingFields;
```

Comment:
`Supplier<Set<FieldId>>` with memoization?

Reply:
Too many fancy words for a performance-related change.
Simple `if (x==null)` will provide the best branch prediction.

Reply:
if `com.google.common.base.Suppliers#memoize` wouldn't be worse, we should use that